### PR TITLE
 reactivate coverage report

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
got the following error:

```bash
In Configurator.php line 152:
                                                                                 
  [Symfony\Component\Config\Definition\Exception\InvalidConfigurationException]  
  coverage_clover XML file is not readable: /home/runner/work/rector/rector/b    
  uild/logs/clover.xml    
```

**Comment by samsonasik** [comment]
It seems something related with the following error: 

```bash

............................................................. 2928 / 2992 ( 97%)

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for <runner@fv-az67.ewrnlpmeiwmuhcy1v0w4ljcxfh.gx.internal.cloudapp.net>) not allowed
```

**Comment by samsonasik** [comment]
It seems the clover xml file actually not generated, there is a warning: 

```bash
Warning:       No filter is configured, code coverage will not be processed
```

**Comment by samsonasik** [comment]
I've added the filter config to phpunit configuration.

File: .github/workflows/code_coverage.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
             -   uses: shivammathur/setup-php@v1
                 with:
                     php-version: 7.3
-                    coverage: xdebug
+                    coverage: pcov
```
Comment (TomasVotruba) [inline .github/workflows/code_coverage.yaml]: :+1:

**Comment by samsonasik** [comment]
@TomasVotruba the coverage is now generated 🎉 , Just different error:

```bash
 Coverage:  73.97% (3240/4380)
Collect git info
Read environment variables
Dump submitting json file: /home/runner/work/rector/rector/build/logs/coveralls-upload.json
File size: 573.98 kB
Submitting to https://coveralls.io/api/v1/jobs
Client error occurred. status: 422 Unprocessable Entity
Couldn't find a repository matching this job.
elapsed time: 0.347 sec memory: 8.00 MB
Error: Process completed with exit code 1.
```

**Comment by samsonasik** [comment]
@TomasVotruba could you verify the `COVERALLS_REPO_TOKEN` settings is correctly configured? Thank you.

**Comment by samsonasik** [comment]
@TomasVotruba It seems really related with `COVERALLS_REPO_TOKEN` settings, please verify if it is correctly configured correctly. Thank you

**Comment by samsonasik** [comment]
Trying with use `coverallsapp/github-action@master` with GITHUB_TOKEN

**Comment by samsonasik** [comment]
@TomasVotruba finally! Coverage uploaded to coveralls:

```bash
Coverage:  73.97% (3240/4380)
Collect git info
Read environment variables
Dump submitting json file: /home/runner/work/rector/rector/build/logs/coveralls-upload.json
File size: 573.96 kB
Submitting to https://coveralls.io/api/v1/jobs
Finish submitting. status: 200 OK
Accepted Job #306493516.1
You can see the build on https://coveralls.io/jobs/69119492
elapsed time: 1.859 sec memory: 8.00 MB
```

see https://coveralls.io/jobs/69119492 

It seems currently only shows `src` directory :

Screen Shot 2020-10-14 at 7 55 11 PM

I am trying to show `packages` and `rules` in next commit.

**Comment by samsonasik** [comment]
@TomasVotruba I've no idea why only `src` only that generated to clover.xml and uploaded to coveralls site. But actually, the process is finally working and uploaded succesfully 🎉 🎉 🎉 

It is ready for merge for now 🎉 🎉 🎉

We eventually need to find out collect multiple directory tests (probably via run tests per-directory inside src, packages, rules separately and then merge them https://github.com/php-coveralls/php-coveralls#phpcov-configuration )

**Pull request merged**